src/traces/ohlc/helpers.js

```diff
@@ -95,14 +95,65 @@ exports.makeTransform = function(traceIn, state, direction) {
 };
 
 exports.getFilterFn = function(direction) {
+    return new _getFilterFn(direction);
+};
+
+function _getFilterFn(direction) {
+    var isPrevThisDirection = null;
+    var cPrev = null;
+    var fn;
+
     switch(direction) {
         case 'increasing':
-            return function(o, c) { return o <= c; };
+            fn = function(o, c) {
+                if(o === c) {
+                    if(c > cPrev) {
+                        return true; // increasing
+                    } else if(c < cPrev) {
+                        return false; // decreasing
+                    } else {
+                        if(isPrevThisDirection === true) {
+                            return true; // determine by last candle
+                        } else if(isPrevThisDirection === false) {
+                            return false; // determine by last candle
+                        } else {
+                            return true; // If we don't have previous data, assume it was increasing
+                        }
+                    }
+                }
+                return o < c;
+            };
+            break;
 
         case 'decreasing':
-            return function(o, c) { return o > c; };
+            fn = function(o, c) {
+                if(o === c) {
+                    if(c > cPrev) {
+                        return false; // increasing
+                    } else if(c < cPrev) {
+                        return true; // decreasing
+                    } else {
+                        if(isPrevThisDirection === true) {
+                            return true; // determine by last candle
+                        } else if(isPrevThisDirection === false) {
+                            return false; // determine by last candle
+                        } else {
+                            return false; // If we don't have previous data, assume it was increasing
+                        }
+                    }
+                }
+                return o > c;
+            };
```

Comment:
This is a lot of duplication with just a few `true` <-> `false` flips... looks like we could keep the perf up without the duplication if we made one function `isIncreasing(o, c)`, which looks at a state var `isPrevIncreasing` (which would also simplify the logic there as we wouldn't need to deal with `null`, just initialize it to `true`), then move the switch logic into creating the wrapper function below:
```
function increasingFilter(o, c) {
    var isThisIncreasing = isIncreasing(o, c);
    isPrevIncreasing = isThisIncreasing; // why add !! to this?
    cPrev = c;
    return isThisIncreasing;
}

function decreasingFilter(o, c) {
    var isThisIncreasing = isIncreasing(o, c);
    isPrevIncreasing = isThisIncreasing; // why add !! to this?
    cPrev = c;
    return !isThisIncreasing; // the only difference...
}
```